src/nvapi_d3d12.cpp

```diff
@@ -285,8 +285,50 @@ extern "C" {
         // Note: One might think that SM here is D3D12 Shader Model, in fact it is the "Streaming Multiprocessor" architecture version
         // Values are valid for Turing and newer only, due to VK_NV_cuda_kernel_launch not being supported by earlier generations
         auto computeCapability = adapter->GetComputeCapability();
-        pGraphicsCaps->majorSMVersion = computeCapability.first;
-        pGraphicsCaps->minorSMVersion = computeCapability.second;
+        if (computeCapability.has_value()) {
+            pGraphicsCaps->majorSMVersion = computeCapability.value().first;
+            pGraphicsCaps->minorSMVersion = computeCapability.value().second;
+        } else if (adapter->HasNvProprietaryDriver()) {
+            // Fallback because older than Turing or no support for VK_NV_cuda_kernel_launch
+            // Based on https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/src/nouveau/winsys/nouveau_device.c
+            switch (adapter->GetArchitectureId()) {
+                case NV_GPU_ARCHITECTURE_GB200:
```

Comment:
What about future architectures that we don't know about yet? Maybe it would make sense to do `arch >= NV_GPU_ARCHITECTURE_*` comparisons then, so e.g. Rubin and Feynman GPUs won't get 0.0 until we handle them explicitly.

Reply:
Yeah, good idea.